lib/runtime-core/src/backend.rs

```diff
@@ -136,6 +136,12 @@ pub struct CompilerConfig {
     /// Enabling this makes execution deterministic but increases runtime overhead.
     pub nan_canonicalization: bool,
 
+    /// Turns off verification that is done by default when `debug_assertions` are enabled
+    /// (for example in 'debug' builds). Enabling this flag will make compilation faster at the
+    /// cost of not detecting bugs in the compiler. The verification steps that this flag
+    /// disables are disabled by default in 'release' builds.
+    pub disable_debug_mode_verification: bool,
```

Comment:
Is there anything stopping us from turning verification on in a release build? Say, for users who are prepared to pay for extra assurance? If we can do that, could we make this control the setting unilaterally, and also remove the negative?

Reply:
Yup, agreed with Nick. Since debug verification is probably always false, we can do something like:

```
    pub enable_ir_verification: bool,
```

(note: this can work for both Cranelift and LLVM, since they both have IR verifiers).

Reply:
I'd like to rename it to something like "extra_safety_checks" or "backend_internal_verification" or something along those lines. There's nothing special about clif/llvm here, singlepass could have its own safety checks too, and it doesn't have an IR.

Reply:
What about `enable_internal_verification` or `enable_verification` (I kind of like keeping backend out, since this might be a setting on the backend config itself?)

Reply:
It's phrased in the negative because we `derive(Default)` which means it defaults to false haha.  I can make a manual implementation of Default though